src/mesh/mesh.cpp

```diff
@@ -930,6 +933,10 @@ void Mesh::Initialize(bool init_problem, ParameterInput *pin, ApplicationInput *
       PARTHENON_REQUIRE_THROWS(
           !(ProblemGenerator != nullptr && block_list[0]->ProblemGenerator != nullptr),
           "Mesh and MeshBlock ProblemGenerators are defined. Please use only one.");
+      PARTHENON_REQUIRE_THROWS(
+          !(PostInitialization != nullptr &&
+            block_list[0]->PostInitialization != nullptr),
+          "Mesh and MeshBlock PostInitializations are defined. Please use only one.");
```

Comment:
Is this actually required? Why can't they both be set if desired?

Reply:
In the current design, if both `Mesh` and `MeshBlock` variations of `PostInitialized` are set, without this `PARTHENON_REQUIRE_THROWS`, only the `Mesh` variation will be called (see the later `if`/`else` logic).  This mimics handling of `ProblemGenerator`.  We could alter the logic to call both, but then there will be an enforced order (which should come first: `Mesh` or `MeshBlock`).  This seems more confusing than beneficial.  IMHO, if you use the `Mesh` version of `ProblemGenerator`, you should use the `Mesh` version of `PostInitialization`. If you use the `MeshBlock` version of `ProblemGenerator`, you should use the `MeshBlock` version of `PostInitialization`.  To me, it seems not favorable to mix and match or to use both.

Reply:
I think allowing two callbacks with the same name is inviting ordering bugs.  It's not obvious from the naming which will get called first, and the order might be especially important during initialization.  We could document it, but why bother when downstreams can control when/how/if they loop through blocks with a few lines in a single MeshData-wide implementation?

Reply:
:+1: that's fine---we can just leave it as @pdmullen designed.